File: src/BinaryTreeInorderTraversal/BinaryTreeInorderTraversal_Iterative.java
class Solution {
    public List<Integer> inorderTraversal(TreeNode root) {
        List<Integer> res = new LinkedList<>();
        //Base case
        if(root == null) return res;
        
        Stack<TreeNode> stack = new Stack<>();
        TreeNode cur = root;
        
        while(cur != null || !stack.isEmpty()){
            if(cur == null){
                TreeNode top = stack.pop();
                res.add(top.val);
                cur = top.right;
            }else{
                stack.add(cur);
                cur = cur.left;
            }
        }
        return res;
    }
}

File: src/Tree/BinaryTreeInorderTraversal/BinaryTreeInorderTraversal_Iterative.py
class Solution:
    def inorderTraversal(self, root: Optional[TreeNode]) -> List[int]:
        res = []
        if root is None:
            return res
        
        cur_node = root
        stack = []
        
        while True:
            while cur_node is not None:
                stack.append(cur_node)
                cur_node = cur_node.left
            
            if len(stack) == 0:
                break
                
            cur_node = stack.pop()
            
            res.append(cur_node.val)
            
            cur_node = cur_node.right
        
        
        return res
